### DISCUSSION TITLE
The handling of language code by i18nextjs is simply incomprehensible

### DISCUSSION THREAD

**Discussion by lindalinya** [i18next/i18next#1813] [General]
![图片](https://user-images.githubusercontent.com/74670117/182072041-ac9bd2b4-4cc9-4052-b0e3-22abd0c081a7.png)
Why i18Next does a hermit conversion to the case of the language code, which may cause the caller's language to not be applied.
On the other hand, when I started lowcaseLNG, it was still possible to register language resources whose language code contained uppercase letters.
In fact, the language code itself is case-insensitive. This strategy doesn't seem very good

**Comment by jamuhl** [comment]
https://www.ietf.org/rfc/rfc4646.txt

**Reply by jamuhl** [reply]
from https://www.fivb.org/VisSDK/VisWebService/LanguageCode.html

**Reply by lindalinya** [reply]
In fact, however, the current implementation is not friendly to all languages in the specification. BTW, the specification explicitly states that the values of the subtags are case-insensitive, and the uniform format is only useful for understanding.

**Reply by jamuhl** [reply]
And what is the issue with using en-US beside you having to strictly use that?
